prime-dotnet-webapi/Services/EnrolleeSubmissionService.cs

```diff
@@ -20,47 +21,50 @@ public class EnrolleeProfileVersionService : BaseService, IEnrolleeProfileVersio
             }
         );
 
-        public EnrolleeProfileVersionService(
+        public EnrolleeSubmissionService(
             ApiDbContext context,
             IHttpContextAccessor httpContext
             ) : base(context, httpContext)
         { }
 
-        public async Task<IEnumerable<EnrolleeProfileVersion>> GetEnrolleeProfileVersionsAsync(int enrolleeId)
+        public async Task<IEnumerable<Submission>> GetEnrolleeSubmissionsAsync(int enrolleeId)
         {
-            return await _context.EnrolleeProfileVersions
+            return await _context.Submissions
                 .Where(epv => epv.EnrolleeId == enrolleeId)
                 .ToListAsync();
         }
 
-        public async Task<EnrolleeProfileVersion> GetEnrolleeProfileVersionAsync(int enrolleeProfileVersionId)
+        public async Task<Submission> GetEnrolleeSubmissionAsync(int enrolleeSubmissionId)
         {
-            return await _context.EnrolleeProfileVersions
-                .SingleOrDefaultAsync(epv => epv.Id == enrolleeProfileVersionId);
+            return await _context.Submissions
+                .SingleOrDefaultAsync(epv => epv.Id == enrolleeSubmissionId);
         }
 
         /**
-          * Get the most recent Profile version before a given time.
+          * Get the most recent submission before a given date.
           */
-        public async Task<EnrolleeProfileVersion> GetEnrolleeProfileVersionBeforeDateAsync(int enrolleeId, DateTimeOffset dateTime)
+        public async Task<Submission> GetEnrolleeSubmissionBeforeDateAsync(int enrolleeId, DateTimeOffset dateTime)
         {
-            return await _context.EnrolleeProfileVersions
+            return await _context.Submissions
             .Where(epv => epv.EnrolleeId == enrolleeId)
             .Where(epv => epv.CreatedDate < dateTime)
             .OrderByDescending(epv => epv.CreatedDate)
             .FirstOrDefaultAsync();
         }
 
-        public async Task CreateEnrolleeProfileVersionAsync(Enrollee enrollee)
+        public async Task CreateEnrolleeSubmissionAsync(Enrollee enrollee)
         {
-            var enrolleeProfileVersion = new EnrolleeProfileVersion
+            AgreementType type = new AgreementEngine().DetermineAgreementType(enrollee).Value;
```

Comment:
This shouldn't be .Value, this will throw if the determined agreement type is null

Reply:
its probably ok to put it all on line 63 as well, don't really need this intermediary variable.